test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/CompleteTagHelperDescriptorComparer.cs

```diff
@@ -23,16 +23,22 @@ bool IEqualityComparer<TagHelperDescriptor>.Equals(TagHelperDescriptor descripto
             return base.Equals(descriptorX, descriptorY) &&
                    // Tests should be exact casing
                    string.Equals(descriptorX.TagName, descriptorY.TagName, StringComparison.Ordinal) &&
-                   descriptorX.Attributes.SequenceEqual(descriptorY.Attributes,
-                                                        CompleteTagHelperAttributeDescriptorComparer.Default);
+                   Enumerable.SequenceEqual(
+                       descriptorX.RequiredAttributes,
+                       descriptorY.RequiredAttributes,
+                       StringComparer.OrdinalIgnoreCase) &&
+                   descriptorX.Attributes.SequenceEqual(
+                       descriptorY.Attributes,
+                        CompleteTagHelperAttributeDescriptorComparer.Default);
```

Comment:
goal is not clear from the code.  further this comparer enforces exact matches that just don't matter e.g. the order of `RequiredAttributes` names and `Attributes` entries.